rfcs/20231025-reusable-kernels/README.md

```diff
@@ -0,0 +1,106 @@
+# Proposal to add a reusable kernel knob
+
+## Introduction
+
+Primitive creation is an expensive operation, which is why oneDNN has a
+primitive cache, a persistent cache, and a kernel cache (for the GPU). For GPU
+implementations, the kernel cache is optional and must be opted into. Although
```

Comment:
My understanding is that the kernel cache is enabled by default because we agreed to make it an internal optimization without exposing it to the user.

Reply:
The kernel cache is enabled by default externally, but the implementations themselves have to opt into the kernel cache API internally. This is a middle-ground solution decided on when developing the kernel cache internal API, but it also helps avoid filling the kernel cache with single-use kernels that are retrieved from the primitive cache already.

Reply:
I see, maybe it makes sense to emphasize that this is about enabling GPU implementations to use the kernel cache i.e. make the implementation kernel cache friendly.